homeassistant/components/sql/config_flow.py

```diff
@@ -0,0 +1,215 @@
+"""Adds config flow for SQL integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import sqlalchemy
+from sqlalchemy.engine import Result
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import scoped_session, sessionmaker
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.recorder import CONF_DB_URL, DEFAULT_DB_FILE, DEFAULT_URL
+from homeassistant.const import CONF_NAME, CONF_UNIT_OF_MEASUREMENT, CONF_VALUE_TEMPLATE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+
+from .const import CONF_COLUMN_NAME, CONF_QUERY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_DB_URL): selector.TextSelector(selector.TextSelectorConfig()),
+        vol.Required(CONF_COLUMN_NAME): selector.TextSelector(
+            selector.TextSelectorConfig()
+        ),
+        vol.Required(CONF_QUERY): selector.TextSelector(selector.TextSelectorConfig()),
+        vol.Optional(CONF_UNIT_OF_MEASUREMENT): selector.TextSelector(
+            selector.TextSelectorConfig()
+        ),
+        vol.Optional(CONF_VALUE_TEMPLATE): selector.TemplateSelector(
+            selector.TemplateSelectorConfig()
+        ),
+    }
+)
+
+
+def validate_sql_select(value: str) -> str | None:
+    """Validate that value is a SQL SELECT query."""
+    if not value.lstrip().lower().startswith("select"):
+        raise ValueError("Incorrect Query")
+    return value
+
+
+def validate_query(db_url: str, query: str, column: str) -> bool:
+    """Validate SQL query."""
+    try:
+        engine = sqlalchemy.create_engine(db_url)
+        sessmaker = scoped_session(sessionmaker(bind=engine))
+    except SQLAlchemyError as error:
+        raise error
+
```

Comment:
In the future, we should set `future=True` for this integration as its the only thing left in the codebase not using the 2.0 api

https://docs.sqlalchemy.org/en/14/changelog/migration_20.html#migration-to-2-0-step-four-use-the-future-flag-on-engine

Reply:
As in change that now or do a separate PR for that?
No idea what it means but I guess todo a separate would make more sense.

Reply:
Future PR for sure.  Example #70358